### ISSUE TITLE
[Bug] select for QueryObserverOptions is typed incorrectly

### ISSUE THREAD

**Issue by QzCurious** [ecyrbe/zodios#196]
The typing of `select` was introduced due to #158

_For convenience, a codesandbox to demo the bug [here](https://codesandbox.io/s/zodios-hooks-select-bug-98wgq2?file=/src/api.ts)._

The bug is described by comments of following code block

```ts
const { data } = apiHook.useUser(undefined, {
  //    ^^^^
  //    data is NOT reflecting to return type of select
  select:
    // ^^ zodios is forcing me follow the type Response
    () => ({
      first_name: "first"
    })
});

function useUser() {
  const { data } = useQuery(["key"], () => ({ name: "name" }), {
    // ^^^^
    // data is reflected to return type of select
    // typed as {
    //   first_name: string;
    // } | undefined
    select: () => ({ first_name: "first" })
  });
}

```

**Comment by ecyrbe** [comment]
Thank you for reporting this issue and taking the Time  to setup a reproductible example.

**Comment by ecyrbe** [comment]
@QzCurious this iw now fixed on `@zodios/react` v10.3.3
